src/python/pants/goal/stats_aggregator.py

```diff
@@ -55,13 +56,32 @@ class StatsAggregatorSubsystem(Subsystem):
         ),
         advanced=True,
     )
+    output_file = StrOption(
+        default=None,
+        metavar="<path>",
+        help="Output the stats to this file. If unspecified, outputs to stdout.",
+    )
+
+
+def _log_or_write_to_file(output_file: Optional[str], text: str) -> None:
+    """Send text to the stdout or write to the output file."""
+    if text:
+        if output_file:
+            with open(output_file, "w") as fh:
```

Comment:
There's a `safe_open` function in `pants.util.dirutil` that might be better, so that someone can specify an output file like `stats/foo.txt` without needing to explicitly create `stats/` directory themselves?

```suggestion
            with safe_open(output_file, "w") as fh:
```